components/chart/editModule/index.tsx

```diff
@@ -0,0 +1,282 @@
+import { Offcanvas } from 'react-bootstrap';
+import { BsX } from "react-icons/bs";
+import React, { PropsWithChildren, useEffect, useMemo, useRef, useState } from 'react';
+import styled from 'styled-components';
+import { NodeType } from '../types';
+import {
+  useUpdateContentInstanceFieldValuesMutation,
+  useContentInstancesQuery,
+  useModelsQuery,
+  ModelFieldsTypes,
+} from '@bcrumbs.net/bc-api';
+import { Controller, useForm, FieldValues } from "react-hook-form";
+import { HYCHART_VIEW_TYPE_ID, TEMPLATE_CONTEXT_ID } from '../Constants';
+import useTagsEnumValuesQuery from '../../../bootstrapers/hychart/utils/useTagsEnumValuesQuery';
+import FieldRenderer from './FieldRenderer';
+type DrawerProps = {
+  module: NodeType;
+  open: boolean;
+  onClose: () => void;
+  lang: string;
+};
+const StyledDrawer = styled.div`
+  position: fixed;
+  padding-left: 20px;
+  padding-right: 20px;
+  top: 38px;
+  bottom: 0;
+  right: 0;
+  max-width: 50vw;
+  width: 25vw;
+  background-color: #eee6dbf7;
+  border-left: 1px solid rgb(100, 57, 0);
+  box-shadow: 0px 0px 25px 0px rgb(100, 57, 0);
+  transition: transform 0.3s ease-in-out, overflow-y 0.3s ease-in-out,
+    max-height 0.3s ease-in-out, padding-bottom 0.3s ease-in-out;
+  z-index: 1050;
+
+  .header {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    direction: ${({ lang }) =>
+    lang === null || lang === 'en' ? 'ltr' : 'rtl'};
+    .title {
+      margin-top: 10px;
+      display: flex;
+      font-weight: bold;
+      align-items: center;
+
+      .linkIcon {
+        width: 20px;
+        height: 20px;
+        cursor: pointer;
+        margin-right: 10px;
+      }
+    }
+
+    .closeIcon {
+      width: 40px;
+      height: 40px;
+      cursor: pointer;
+    }
+  }
+
+  &.show {
+    transform: translateX(0);
+    overflow-y: auto;
+    max-height: 100%;
+    padding-bottom: 10px;
+  }
+`;
+const SuccessToast = styled.div`
+  position: fixed;
+  bottom: 20px;
+  left: 64%;
+  transform: translateX(-50%);
+  background-color: #4caf50; 
+  color: #fff;
+  padding: 10px 20px;
+  border-radius: 5px;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.5);
+  z-index: 9999;
+`;
+
+const ErrorToast = styled.div`
+  position: fixed;
+  bottom: 20px;
+  left: 50%;
+  transform: translateX(-50%);
+  background-color: #ff6347;
+  color: #fff;
+  padding: 10px 20px;
+  border-radius: 5px;
+  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.2);
+  z-index: 9999;
+`;
```

Comment:
Let's define these a generic styles, maybe we can start to put them in
packages\hychart\components\common\toasts.tsx